client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -46,11 +41,11 @@ export default class EmbedViewManager extends EventEmitter {
 		}
 	}
 
-	addListener( event, listener ) {
+	addListener( event, listener, store ) {
 		super.addListener( event, listener );
-
 		if ( 'change' === event && 1 === this.listeners( event ).length ) {
-			sites.addListener( event, this.sitesListener );
+			this.store = store;
+			this.unsubscribeStore = store.subscribe( this.sitesListener );
```

Comment:
Hi @gwwar, thank you a lot for your explanation, I think I have the information I need to give it a try. I will comment when I have this alternative approach ready :)